### PR TITLE
Fixes issue where QR code was "recalculated" and rendered each frame.

### LINKED ISSUE #31: QR-code / AP view in onboarding has low FPS
Drops to 12 from 20. goes back to 20 when clicking out.

### PR DESCRIPTION
**Pull request by carlhampuswall**
Fixes #31

### CODE REVIEW

**Comment by carlhampuswall** [comment]
This one can be made to work in a better way in the future, but does resolve key issue for now.

File: firmware/src/error_handling_flow/error_handling_flow.cpp [hunk 1]
```diff
@@ -15,16 +35,34 @@ void ErrorHandlingFlow::handleEvent(WiFiEvent event)
...
     {
-    case SK_MQTT_CONNECTION_FAILED:
     case SK_MQTT_RETRY_LIMIT_REACHED:
+        if (!WiFi.isConnected())
+        {
+            sprintf(ip_data, "%s", "http://192.168.4.1/mqtt"); // always the same
+        }
+        else
+        {
+            sprintf(ip_data, "http://%s/mqtt", WiFi.localIP().toString().c_str());
```
Comment (brushknight) [inline firmware/src/error_handling_flow/error_handling_flow.cpp:45]: In theory, we should not access global objects like that. But we don't have a good mechanism to pass this data around yet.

Comment (carlhampuswall) [reply firmware/src/error_handling_flow/error_handling_flow.cpp:45]: WiFi.localIP()?

File: firmware/src/onboarding_flow/onboarding_flow.cpp [hunk 2]
```diff
@@ -1,49 +1,8 @@
 #include "onboarding_flow.h"
 #include "apps/icons.h"
-#include "qrcode.h"
 
-OnboardingFlow::OnboardingFlow()
+OnboardingFlow::OnboardingFlow(TFT_eSprite *spr_, TFT_eSprite qrcode_spr_) : spr_(spr_), qrcode_spr_(qrcode_spr_)
 {
-
-    root_level_motor_config = PB_SmartKnobConfig{
```
Comment (brushknight) [inline firmware/src/onboarding_flow/onboarding_flow.cpp:8]: Why did you remove this?

Comment (brushknight) [reply firmware/src/onboarding_flow/onboarding_flow.cpp:8]: nevermind, found my answer

File: firmware/src/onboarding_flow/onboarding_flow.cpp [hunk 3]
```diff
@@ -112,23 +71,61 @@ EntityStateUpdate OnboardingFlow::update(AppState state)
...
         is_wifi_ap_client_connected = event.body.ap_client.connected;
         if (is_wifi_ap_client_connected)
         {
+            if (is_wifi_ap_client_connected)
+            {
+                sprintf(ip_data, "%s", "http://192.168.4.1"); // always the same
+            }
+            else
+            {
+                sprintf(ip_data, "%s", "http://10.0.0.1"); // TODO: fetch real ip address
```
Comment (brushknight) [inline firmware/src/onboarding_flow/onboarding_flow.cpp:125]: You should fix this todo ;)

File: firmware/src/onboarding_flow/onboarding_flow.cpp (hunk 2 above)
Comment (carlhampuswall) [reply firmware/src/onboarding_flow/onboarding_flow.cpp:8]: TFT_eSprite doesnt have a default constructor and either needs a reference to tft_ (TFT_eSpi) or to be initialized from the start.

**Review by brushknight — COMMENTED** [review]
One small TODO needs to be fixed

File: firmware/src/onboarding_flow/onboarding_flow.cpp (hunk 3 above)
Comment (carlhampuswall) [reply firmware/src/onboarding_flow/onboarding_flow.cpp:125]: Ur todo :P 

Will fix :+1:

**Review by brushknight — APPROVED** [review]

**Pull request merged**